src/mopidy_spotify/playlists.py

```diff
@@ -13,6 +15,8 @@ def __init__(self, backend):
         self._timeout = self._backend._config["spotify"]["timeout"]
         self._loaded = False
 
+        self._refreshing = False
```

Comment:
I would use the lock as jodal suggested, otherwise you'll need to have a something protecting `_refreshing` from concurrent access. And if you don't go that route I would annotate with comments saying something about what lock needs to be held for it to be safe to read/write this variable.

(This might be all my C++ at work bleeding over, but the concurrency issues should still be the same)

Reply:
```
    def __init__(self, backend):
        self._refresh_mutex = threading.Lock()

    def refresh(self):
        if not self._refresh_mutex.acquire(blocking=False):
            logger.info("Refreshing Spotify playlists already in progress")
            return
        ...

    def _refresh_tracks(self, playlist_uris):
        if not self._refresh_mutex.locked():
            logger.error("Lock must be held before calling this method")
            return

        try:
            ...
        except Exception:
            logger.exception("Error occurred while refreshing Spotify playlists tracks")
        finally:
            self._refresh_mutex.release()
```

Reply:
I was being lazy and relying on the fact this is atomic in (current versions of) Python. I should just use a lock and make this interaction explicit.